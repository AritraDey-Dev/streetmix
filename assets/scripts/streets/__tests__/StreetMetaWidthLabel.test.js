/* eslint-env jest */
import React from 'react'
import { fireEvent } from '@testing-library/react'
import { renderWithIntl } from '../../../../test/helpers/render'
import StreetMetaWidthLabel from '../StreetMetaWidthLabel'

const dummyStreetObject = {
  units: 0,
  width: 10,
  occupiedWidth: 10,
  remainingWidth: 0
}

describe('StreetMetaWidthLabel', () => {
  it('renders when editable', () => {
    const wrapper = renderWithIntl(
      <StreetMetaWidthLabel
        street={dummyStreetObject}
        editable
        onClick={jest.fn()}
      />
    )

    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('renders when not editable', () => {
    const wrapper = renderWithIntl(
      <StreetMetaWidthLabel
        street={dummyStreetObject}
        editable={false}
        onClick={jest.fn()}
      />
    )

    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('renders with remaining width', () => {
    const wrapper = renderWithIntl(
      <StreetMetaWidthLabel
        street={{
          ...dummyStreetObject,
          occupiedWidth: 9,
          remainingWidth: 1
        }}
        editable
        onClick={jest.fn()}
      />
    )

    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('renders with over available width', () => {
    const wrapper = renderWithIntl(
      <StreetMetaWidthLabel
        street={{
          ...dummyStreetObject,
          occupiedWidth: 11,
          remainingWidth: -1
        }}
        editable
        onClick={jest.fn()}
      />
    )

    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('calls onClick handler when clicked', () => {
    const handleClick = jest.fn()
    const wrapper = renderWithIntl(
      <StreetMetaWidthLabel
        street={dummyStreetObject}
        editable
        onClick={handleClick}
      />
    )

    fireEvent.click(wrapper.getByTitle('Change width of the street'))
    expect(handleClick).toBeCalled()
  })
})
