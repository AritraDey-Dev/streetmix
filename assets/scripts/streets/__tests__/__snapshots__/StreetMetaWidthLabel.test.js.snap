// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StreetMetaWidthLabel renders when editable 1`] = `
<span
  className="street-width street-width-editable"
  onClick={[MockFunction]}
  title="Change width of the street"
>
  <injectIntl(FormattedMessage)
    defaultMessage="{width} width"
    id="width.label"
    values={
      Object {
        "width": "3 m",
      }
    }
  />
</span>
`;

exports[`StreetMetaWidthLabel renders when not editable 1`] = `
<span
  className="street-width"
  onClick={[MockFunction]}
  title={null}
>
  <injectIntl(FormattedMessage)
    defaultMessage="{width} width"
    id="width.label"
    values={
      Object {
        "width": "3 m",
      }
    }
  />
</span>
`;

exports[`StreetMetaWidthLabel renders with over available width 1`] = `
<span
  className="street-width street-width-editable"
  onClick={[MockFunction]}
  title="Change width of the street"
>
  <injectIntl(FormattedMessage)
    defaultMessage="{width} width"
    id="width.label"
    values={
      Object {
        "width": "3 m",
      }
    }
  />
  <span
    className="street-width-over"
  >
    <injectIntl(FormattedMessage)
      defaultMessage="({width} over)"
      id="width.over"
      values={
        Object {
          "width": "0.3 m",
        }
      }
    />
  </span>
</span>
`;

exports[`StreetMetaWidthLabel renders with remaining width 1`] = `
<span
  className="street-width street-width-editable"
  onClick={[MockFunction]}
  title="Change width of the street"
>
  <injectIntl(FormattedMessage)
    defaultMessage="{width} width"
    id="width.label"
    values={
      Object {
        "width": "3 m",
      }
    }
  />
  <span
    className="street-width-under"
  >
    <injectIntl(FormattedMessage)
      defaultMessage="({width} room)"
      id="width.under"
      values={
        Object {
          "width": "0.3 m",
        }
      }
    />
  </span>
</span>
`;
